### PR TITLE
Add Slack's Source 

### PR DESCRIPTION
**Pull request by yevhenii-ldv**
## What
*Add new source [Slack API](https://api.slack.com/) using [Singel Tap](https://github.com/singer-io/tap-slack).*

## How
*Describe the solution*

## Checklist
- [ ] *Create config API model*
- [ ] *Implement config persistence*
- [ ] *Add persistence tests*

## Recommended reading order
1. `test.java`
1. `component.ts`
1. the rest

### CODE REVIEW

File: airbyte-integrations/connectors/source-slack-singer/source_slack_singer/spec.json [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+      "join_public_channels": {
+        "title": "Join Public channels",
+        "type": "boolean",
+        "description": "This option joins all public channels in your organization.",
+        "default": true
+      },
+      "exclude_archived": {
+        "title": "Exclude Archived channels",
+        "type": "boolean",
+        "description": "You are adjusting the synchronization of data from archive channels.",
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-slack-singer/source_slack_singer/spec.json:41]:
```suggestion
        "description": "Whether to exclude information about archived channels from syncs",
```

File: airbyte-integrations/connectors/source-slack-singer/source_slack_singer/spec.json [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      "private_channels": {
+        "title": "Private channels",
+        "type": "boolean",
+        "description": "This option adds Slack private channels for synchronization.",
+        "default": false
+      },
+      "join_public_channels": {
+        "title": "Join Public channels",
+        "type": "boolean",
+        "description": "This option joins all public channels in your organization.",
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-slack-singer/source_slack_singer/spec.json:35]:
```suggestion
        "description": "Whether to sync data from all public channels in your organization.",
```

File: airbyte-integrations/connectors/source-slack-singer/source_slack_singer/spec.json [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        "pattern": "^[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}Z$"
+      },
+      "token": {
+        "type": "string",
+        "description": "The Slack API token to interact with your Slack workspace."
+      },
+      "private_channels": {
+        "title": "Private channels",
+        "type": "boolean",
+        "description": "This option adds Slack private channels for synchronization.",
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-slack-singer/source_slack_singer/spec.json:29]:
```suggestion
        "description": "Whether to sync data from private channels.",
```

**Review by sherifnada — APPROVED** [review]
LGTM! some small changes needed to docs but will merge in the morning if they are in by then.

**Comment by yevhenii-ldv** [comment]
Hi!) 
File `spec.json` updated as per your comments

**Pull request merged**